src/transformers/trainer_utils.py

```diff
@@ -499,6 +499,18 @@ def stop_and_update_metrics(self, metrics=None):
             self.update_metrics(stage, metrics)
 
 
+def has_length(dataset):
+    """
+    Checks if the dataset implements __len__() and it doesn't raise an error
+    """
+    try:
+        if len(dataset) is not None:
+            return True
```

Comment:
```suggestion
        return len(dataset) is not None
```
I think we will return `None` instead of `False` when the length is None otherwise.